command/command.c

```diff
@@ -289,21 +295,67 @@ process_command_output (CommandApplet *command_applet, gchar *output)
     }
 }
 
+static void
+subprocess_callback (GObject *source, GAsyncResult *result, gpointer data)
+{
+    GSubprocess *subprocess;
+    CommandApplet *command_applet;
+    GInputStream *in;
+    gsize bytes_read;
+    gchar *output;
+    GError *error = NULL;
+
+    command_applet = (CommandApplet *) data;
+    subprocess = command_applet->subprocess;
+
+    if (!g_subprocess_wait_finish (subprocess, result, &error))
+    {
+        g_clear_error (&error);
+        return;
+    }
+
+    output = g_new0 (gchar, command_applet->width);
+    in = g_subprocess_get_stdout_pipe (subprocess);
+    g_input_stream_read_all (in, output, command_applet->width, &bytes_read, NULL, NULL);
+
+    process_command_output (command_applet, output);
+    g_free (output);
+
+    if (command_applet->subprocess != NULL) {
+        g_object_unref (command_applet->subprocess);
+        command_applet->subprocess = NULL;
+    }
+}
+
 static gboolean
 command_execute (CommandApplet *command_applet)
 {
     GError *error = NULL;
-    gchar *output = NULL;
-    gint ret = 0;
+    gint argc;
+    gchar **argv;
 
-    if (g_spawn_command_line_sync (command_applet->command, &output, NULL, &ret, &error))
+    if (! g_shell_parse_argv (command_applet->command, &argc, &argv, &error))
     {
-        process_command_output (command_applet, output);
+        gtk_label_set_text (command_applet->label, ERROR_OUTPUT);
+        g_clear_error (&error);
+        return FALSE;
     }
-    else
+
+    if (command_applet->subprocess != NULL && G_IS_SUBPROCESS(command_applet->subprocess)) {
```

Comment:
`G_IS_SUBPROCESS` is not needed here! I think these functions are meant to be used in public API function to check if API user pass correct arguments to functions.